src/shared-components/accordion/index.tsx

```diff
@@ -27,7 +27,12 @@ type AccordionProps = {
   title: React.ReactNode;
 };
 
-class Accordion extends React.Component<
+/**
+ * A list of items that allows each item's content to be expanded and collapsed by clicking its title bar.
+ *
+ * The accordion component expands to reveal hidden information. They should be used when you need to fit a large amount of content but don't want to visually overwhelm the user.
+ */
+export class Accordion extends React.Component<
```

Comment:
One detail I noticed while working on this is that, for TypeScript files in particular, only Named Exports allowed us to actually document the type/prop information within a given story. Because of this, this PR **overhauls our existing import/export structure for components, and prefers named exports in almost every instance**. The sole exception is sub-component usage that we do not document (e.g. the `Arrow` component for an `Accordion`) or exports that make use of `withDeprecationWarning` like `Button` and `RoundButton`. 

These changes are reflected in `src/shared-components/index.ts`, and I updated the `.size-snapshot.json` file to show that the effect on our bundle is neglible. The majority of the size changes comes from shipping the expanded JSDoc comments, which wouldn't matter when passing it through a minifier that removes comments anyway. Additionally, this PR also removes `defaultProps` in favor of ES6 defaults for function components, so some components actually shrink. Those changes may conflict with #423, but since this PR probably won't get merged until sometime next week, I will just deal with the merge conflicts that result from those changes (if any).